src/intro_movie_scene_5.c

```diff
@@ -0,0 +1,239 @@
+#include "intro_movie_internal.h"
+#include "system.h"
+#include "unk_0200B150.h"
+#include "unk_0200FA24.h"
+
+enum IntroScene5State {
+    INTRO_SCENE5_WIPE_IN,
+    INTRO_SCENE5_WAIT_WIPE,
+    INTRO_SCENE5_BEGIN_BG_SCROLL,
+    INTRO_SCENE5_WAIT_BG_SCROLL,
+    INTRO_SCENE5_WAIT_FADE_OUT,
+};
+
+static void IntroMovie_Scene5_VBlankCB(void *pVoid);
+static void IntroMovie_Scene5_Init(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData);
+static BOOL IntroMovie_Scene5_Main(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData, int totalFrames);
+static void IntroMovie_Scene5_Exit(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData);
+static void IntroMovie_Scene5_InitBgs(IntroMovieOvyData *data);
+static void InroMovie_Scene5_LoadBgGfx(BgConfig *bgConfig, IntroMovieScene5Data *sceneData);
+static void IntroMovie_Scene5_EnableBgLayers(IntroMovieScene5Data *sceneData, BgConfig *bgConfig);
+static void IntroMovie_Scene5_LoadSpriteGfx(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData);
+static void IntroMovie_Scene5_DestroySprites(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData);
+static void IntroMovie_Scene5_CreateSprites(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData);
+
+BOOL IntroMovie_Scene5(IntroMovieOvyData *data, void *pVoid) {
+    IntroMovieScene5Data *sceneData = (IntroMovieScene5Data *)pVoid;
+
+    if (IntroMovie_GetIntroSkippedFlag(data)) {
+        sceneData->state = 2;
+    }
+
+    switch (sceneData->state) {
+    case INTRO_MOVIE_SCENE_INIT:
+        IntroMovie_Scene5_Init(data, sceneData);
+        ++sceneData->state;
+        break;
+    case INTRO_MOVIE_SCENE_RUN:
+        if (IntroMovie_Scene5_Main(data, sceneData, IntroMovie_GetTotalFrameCount(data))) {
+            ++sceneData->state;
+        }
+        break;
+    case INTRO_MOVIE_SCENE_CLEANUP:
+        IntroMovie_Scene5_Exit(data, sceneData);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void IntroMovie_Scene5_VBlankCB(void *pVoid) {
+    IntroMovieOvyData *data = (IntroMovieOvyData *)pVoid;
+
+    DoScheduledBgGpuUpdates(IntroMovie_GetBgConfig(data));
+    OamManager_ApplyAndResetBuffers();
+}
+
+static void IntroMovie_Scene5_Init(IntroMovieOvyData *data, IntroMovieScene5Data *sceneData) {
+    BgConfig *bgConfig = IntroMovie_GetBgConfig(data);
+    gSystem.screensFlipped = TRUE;
+    GfGfx_SwapDisplay();
+    sub_0200FBF4(PM_LCD_TOP, RGB_BLACK);
+    sub_0200FBF4(PM_LCD_BOTTOM, RGB_BLACK);
+    IntroMovie_Scene5_InitBgs(data);
+    IntroMovie_InitBgAnimGxState(data);
+    Main_SetVBlankIntrCB(IntroMovie_Scene5_VBlankCB, data);
+    InroMovie_Scene5_LoadBgGfx(bgConfig, sceneData);
+    IntroMovie_RendererSetSurfaceCoords(data, 0, 0, 0, 256);
+    IntroMovie_Scene5_LoadSpriteGfx(data, sceneData);
+    IntroMovie_Scene5_CreateSprites(data, sceneData);
+    IntroMovie_Scene5_EnableBgLayers(sceneData, bgConfig);
+    sceneData->needFreeGfx = TRUE;
+}
+
+#ifdef HEARTGOLD
+#define BG_SCROLL_SPEED -0x40
```

Comment:
this is a version difference?

Reply:
Yep! At the end of the intro, some bg layers scroll down in heartgold (to simulate the camera tilting up) or up in soulsilver (camera tilting down)